driver/node.go

```diff
@@ -226,11 +260,34 @@ func (s *nodeService) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 
 	// remove device file if target_path is device, unmount target_path otherwise
 	if info.IsDir() {
-		return s.nodeUnpublishFilesystemVolume(req, device)
+		unpublishResp, err := s.nodeUnpublishFilesystemVolume(req, device)
+		if err != nil {
+			return unpublishResp, err
+		}
+		listResp, err := s.client.GetLVList(ctx, &proto.Empty{})
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to list LV: %v", err)
+		}
+		volume := s.findVolumeByID(listResp, req.GetVolumeId())
+		if volume != nil && s.isEphemeralVolume(volume) {
+			if _, err = s.lvService.RemoveLV(ctx, &proto.RemoveLVRequest{Name: req.GetVolumeId()}); err != nil {
+				return nil, status.Errorf(codes.Internal, "Failed to remove LV for %s: %v", req.GetVolumeId(), err)
+			}
+		}
+		return &csi.NodeUnpublishVolumeResponse{}, nil
```

Comment:
I think this is more appropriate:
```suggestion
		return unpublishResp, nil
```

Reply:
This value came from much earlier in the function before a lot of processing including getting  occurred, so I don't understand how reusing this value makes sense. Can you elaborate?

I realize this doesn't matter a lot since they end up being the same thing, but philosophically, it seems odd to me.

Reply:
True.  marked as resolved.